engine/api/event_v2/event_analysis.go

```diff
@@ -2,33 +2,44 @@ package event_v2
 
 import (
 	"context"
+	"encoding/json"
 
 	"github.com/ovh/cds/engine/cache"
 	"github.com/ovh/cds/sdk"
 )
 
-func PublishAnalysisStart(ctx context.Context, store cache.Store, vcsName, repoName string, a *sdk.ProjectRepositoryAnalysis) {
+func PublishAnalysisStart(ctx context.Context, store cache.Store, vcsName, repoName string, a *sdk.ProjectRepositoryAnalysis, u *sdk.AuthentifiedUser) {
+	bts, _ := json.Marshal(a)
 	e := sdk.EventV2{
 		ID:         sdk.UUID(),
 		ProjectKey: a.ProjectKey,
 		VCSName:    vcsName,
 		Repository: repoName,
 		Status:     a.Status,
 		Type:       sdk.EventAnalysisStart,
-		Payload:    *a,
+		Payload:    bts,
+	}
+	if u != nil {
```

Comment:
if the user can be nil if some case, we should never set it.

event data must be reliable